File: tests/fixtures/sample_js/service.js
import { User } from "./models.js";
import { UserRepository } from "./repository.js";

export class UserService {
  constructor() {
    this.repo = new UserRepository();
  }

  register(name, email) {
    const user = new User(name, email);
    this.repo.add(user);
    return user;
  }

  headcount() {
    return this.repo.count();
  }
}

/** Build a default service instance. */
export const makeDefaultService = () => new UserService();

File: tests/fixtures/sample_go/service.go
package sample

// UserService coordinates user operations.
type UserService struct {
	repo *UserRepository
}

func (s *UserService) Register(name, email string) *User {
	u := &User{Name: name, Email: email}
	s.repo.Add(u)
	return u
}

func (s *UserService) Headcount() int {
	return s.repo.Count()
}

// MakeDefaultService builds a service with a fresh repository.
func MakeDefaultService() *UserService {
	return &UserService{repo: NewUserRepository()}
}
